src/main/java/dev/slimevr/vr/processor/skeleton/LegTweakBuffer.java

```diff
@@ -0,0 +1,484 @@
+		return leftLegState;
+	}
+
+	public void setLeftLegState(int leftLegState) {
+		this.leftLegState = leftLegState;
+	}
+
+	public int getRightLegState() {
+		return rightLegState;
+	}
+
+	public void setRightLegState(int rightLegState) {
+		this.rightLegState = rightLegState;
+	}
+
+	public void setParent(LegTweakBuffer parent) {
+		this.parent = parent;
+	}
+
+	public LegTweakBuffer getParent() {
+		return parent;
+	}
+
+	public void setLeftFootAcceleration(Vector3f leftFootAcceleration) {
+		this.leftFootAcceleration = leftFootAcceleration.clone();
+	}
+
+	public void setRightFootAcceleration(Vector3f rightFootAcceleration) {
+		this.rightFootAcceleration = rightFootAcceleration.clone();
+	}
+
+	public void getLeftFootAcceleration(Vector3f leftFootAcceleration) {
+		leftFootAcceleration.set(this.leftFootAcceleration);
+	}
+
+	public void getRightFootAcceleration(Vector3f rightFootAcceleration) {
+		rightFootAcceleration.set(this.rightFootAcceleration);
+	}
+
+	public float getLeftFootAccelerationMagnitude() {
+		return this.leftFootAcceleration.length();
+	}
+
+	public float getRightFootAccelerationMagnitude() {
+		return this.rightFootAcceleration.length();
+	}
+
+	public float getLeftFootAccelerationY() {
+		return this.leftFootAcceleration.y;
+	}
+
+	public float getRightFootAccelerationY() {
+		return this.rightFootAcceleration.y;
+	}
+
+	public void setDetectionMode(int mode) {
+		this.detectionMode = mode;
+	}
+
+	// calculate momvent attributes
+	public void calculateFootAttributes(boolean active) {
+		updateFrameNumber(0);
+
+		// compute attributes of the legs
+		computeVelocity();
+		computeAccelerationMagnitude();
+
+		// check if the acceleration triggers forced unlock
+		if (detectionMode == FOOT_ACCEL) {
+			computeAccelerationAboveThresholdFootTrackers();
+		} else {
+			computeAccelerationAboveThresholdAnkleTrackers();
+		}
+
+		// calculate the scalar for other parameters
+		leftFootSensitivity = getLeftFootScalar();
+		rightFootSensitivity = getRightFootScalar();
+
+		// if correction is inactive state is unknown (default to unlocked)
+		if (!active) {
+			leftLegState = UNLOCKED;
+			rightLegState = UNLOCKED;
+		} else {
+			computeState();
+		}
+	}
+
+	// update the frame number of all the frames
+	public void updateFrameNumber(int frameNumber) {
+		this.frameNumber = frameNumber;
+		if (this.frameNumber >= 10) {
+			this.parent = null; // once a frame is 10 frames old, it is no
+								// longer
+								// needed
+		}
+		if (parent != null) {
+			parent.updateFrameNumber(frameNumber + 1);
+		}
+	}
+
+	// compute the state of the legs
+	private void computeState() {
+		// based on the last state of the legs compute their state for this
+		// individual frame
+		leftLegState = checkStateLeft();
+
+		rightLegState = checkStateRight();
+	}
+
+	// check if a locked foot should stay locked or be released
+	private int checkStateLeft() {
+		float timeStep = 1.0f / ((timeOfFrame - parent.timeOfFrame) / 1000000000.0f);
```

Comment:
Can we have a comment or constant explaining `1000000000.0f`? It looks like converting from nanoseconds to seconds, but some clarification could be nice.